datadog_lambda/wrapper.py

```diff
@@ -102,34 +105,42 @@ def __init__(self, func):
 
     def __call__(self, event, context, **kwargs):
         """Executes when the wrapped function gets called"""
-        self._before(event, context)
+        trigger_tags = extract_trigger_tags(event, context)
+        self._before(event, context, trigger_tags)
+        response = None
         try:
-            return self.func(event, context, **kwargs)
+            response = self.func(event, context, **kwargs)
+            return response
         except Exception:
             submit_errors_metric(context)
             if self.span:
                 self.span.set_traceback()
             raise
         finally:
-            self._after(event, context)
+            self._after(event, context, response, trigger_tags)
 
-    def _before(self, event, context):
+    def _before(self, event, context, trigger_tags):
         try:
 
             set_cold_start()
             submit_invocations_metric(context)
-            # Extract Datadog trace context from incoming requests
-            dd_context = extract_dd_trace_context(event, context)
+            # Extract Datadog trace context and source from incoming requests
+            dd_context, trace_context_source = extract_dd_trace_context(event, context)
+            # Create a Datadog X-Ray subsegment with the trace context
+            create_dd_dummy_metadata_subsegment(
```

Comment:
Related to the previous comment. I think you should call this function `if dd_context and trace_context_source == TraceContextSource.EVENT`?